robot/Cumulus/perftests/BDI_Tests.robot

```diff
@@ -1,138 +1,142 @@
+## @description: NPSP BDI Performance Tests
+##               Two Separate BDI Jobs, each with 10K DI records. One is 100% match to existing Contacts, and the other has 0 matches.
+## @date 2021-08-23
+
 *** Variables ***
-${database_url} =    
-${persistent_org} =     ${False}
+${persistent_org} =         ${False}
+${AdvMappingConfigValue} =  Data Import Field Mapping
+${DataImportBatchSize} =    250
+${DataImportRecordCount} =  10000
+${RecipeWithNoMatches} =    datasets/bdi_benchmark/BDI-with-matching.recipe.yml
+${RecipeWithMatches} =      datasets/bdi_benchmark/BDI-without-matching.recipe.yml
 
 *** Settings ***
 
 Resource  cumulusci/robotframework/CumulusCI.robot
 Resource        robot/Cumulus/resources/NPSP.robot
 Resource        robot/Cumulus/resources/Data.robot
 Resource        robot/Cumulus/resources/BDI_API.robot
-Suite Setup       Workaround Bug
+Suite Setup     Configure Org
 
 *** Keywords ***
-Clear Generated Records
-    Output  Clearing Generated Records
-    # Organized in dependency order
-    Bulk Delete     DataImport__c, CustomObject1__c, CustomObject2__c, CustomObject3__c      
-    Bulk Delete     Account_Soft_Credit__c     where=Opportunity__r.Primary_Contact__r.LastName Like '%BDITEST%' OR Account__r.Name Like '%BDITEST%'
-    Bulk Delete     Allocation__c        where=Opportunity__r.Primary_Contact__r.LastName Like '%BDITEST%' OR Opportunity__r.Account.Name Like '%BDITEST%'
-    Bulk Delete     npe01__OppPayment__c     where=npe01__Opportunity__r.Primary_Contact__r.LastName Like '%BDITEST%' OR npe01__Opportunity__r.Account.Name Like '%BDITEST%'
-    Bulk Delete     Opportunity     where=Primary_Contact__r.LastName Like '%BDITEST%' OR Account.Name Like '%BDITEST%'
-    Bulk Delete     Account     where=Name Like '%BDITEST%'
-    Bulk Delete     Contact     where=LastName Like '%BDITEST%'
+## =============================================================================================
+## Main Testing Keywords:
+## =============================================================================================
 
+Setup For Test
+    [Arguments]             ${data-matching}
+    Output      \n
+    Output      Preparing Data for Test
+    Clear Test Data
+    Generate Data           ${data-matching}
+    Output      Executing Data Import Batch Job
+
+Configure Org
+    Disable Duplicate Matching
+    Delete Default Data
+    Configure BDI       ${AdvMappingConfigValue}
+    Ensure Custom Metadata Was Deployed
+
+## =============================================================================================
+## Data Creation Keywords:
+## =============================================================================================
 Generate Data
-    [Arguments]    ${count}
-    ${count} =  Convert To Integer	${count}
+    [Arguments]    ${data-matching}
+    ${count} =  Convert To Integer      ${DataImportRecordCount}
+
+    ${recipe} =     Set Variable if    '${data-matching}'=='Full-Match'       ${RecipeWithMatches}
```

Comment:
@prescod  Just re-tested. The full match test is using the wrong recipe file, which means this line is not returning the expected value. I'll try to figure out why, unless you see something obvious.

Reply:
@prescod  I added the `ELSE`, but it hasn't helped.  This is what I see from the  output below:
> Generating 10 Records for 'Full-Match' Test Using 'datasets/bdi_benchmark/BDI-without-matching.recipe.yml'

Any thoughts?